addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactP20Handler.java

```diff
@@ -0,0 +1,967 @@
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_U1),
+							newState);
+					break;
+				}
+				case "U2": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_U2),
+							newState);
+					break;
+				}
+				case "U3": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_U3),
+							newState);
+					break;
+				}
+				case "I1": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state / 1000);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_I1),
+							newState);
+					break;
+				}
+				case "I2": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state / 1000);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_I2),
+							newState);
+					break;
+				}
+				case "I3": {
+					int state = entry.getValue().getAsInt();
+					State newState = new DecimalType(state / 1000);
+					updateState(
+							new ChannelUID(getThing().getUID(), CHANNEL_I3),
+							newState);
+					break;
+				}
+				case "P": {
+					long state = entry.getValue().getAsLong();
+					State newState = new DecimalType(state / 1000);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_POWER), newState);
+					break;
+				}
+				case "PF": {
+					int state = entry.getValue().getAsInt();
+					State newState = new PercentType(state / 10);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_POWER_FACTOR), newState);
+					break;
+				}
+				case "E pres": {
+					long state = entry.getValue().getAsLong();
+					State newState = new DecimalType(state / 10);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_SESSION_CONSUMPTION), newState);
+					break;
+				}
+				case "E total": {
+					long state = entry.getValue().getAsLong();
+					State newState = new DecimalType(state / 10);
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_TOTAL_CONSUMPTION), newState);
+					break;
+				}
+				}
+			}
+
+		} catch (JsonParseException e) {
+			logger.warn("Some invalid JSON data will be ignored: '{}'",
+					response);
+		}
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+
+		switch (channelUID.getId()) {
+		case CHANNEL_MAX_PRESET_CURRENT: {
+			if (command instanceof DecimalType) {
+				sendCommand("curr "
+						+ String.valueOf(Math.min(
+								Math.max(6000,
+										Integer.parseInt(command.toString())),
+								maxSystemCurrent)));
+			}
+			break;
+		}
+		case CHANNEL_MAX_PRESET_CURRENT_RANGE: {
+			if (command instanceof OnOffType
+					|| command instanceof IncreaseDecreaseType
+					|| command instanceof PercentType) {
+
+				int newValue = 6000;
+				if (command instanceof IncreaseDecreaseType
```

Comment:
You do not need the check "command instanceof IncreaseDecreaseType" if you are doing a "command == IncreaseDecreaseType.INCREASE" afterwards anyhow